src/Service.Tests/CosmosTests/SchemaGeneratorFactoryTests.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Generator;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Azure.DataApiBuilder.Service.Tests.CosmosTests
+{
+    /// <summary>
+    /// Contains unit tests for SchemaGeneratorFactory related to Cosmos DB NoSQL.
+    /// </summary>
+    [TestClass, TestCategory(TestCategory.COSMOSDBNOSQL)]
+    public class SchemaGeneratorFactoryTests
+    {
+        /// <summary>
+        /// Tests the ExportGraphQLFromCosmosDB method to ensure it generates a GraphQL schema successfully from Cosmos DB.
+        /// </summary>
+        [TestMethod]
+        public async Task ExportGraphQLFromCosmosDB_GeneratesSchemaSuccessfully()
+        {
+            // Arrange: Set up test configuration, mocks, and test data.
+
+            // Runtime configuration for the schema generation.
+            RuntimeConfig runtimeConfig = new(
+               Schema: "schema",
+               DataSource: new DataSource(DatabaseType.CosmosDB_NoSQL, "dummy-connection-string", new()
```

Comment:
consider saying noop-connection-string instead of dummy - policheck will complain